### ISSUE TITLE
Add Images

### ISSUE THREAD

**Issue by Pablo-Martinetz** [webosbrew/custom-screensaver#11]
Wehre can i add my custom images for this screen safer

**Comment by noromamai** [comment]
the DVD image (png) is base64 encoded into the screensaver-main.qml file. don't forget to change the size of width and height to match your image. image should be black and transparant. transparant areas are what will change color

**Comment by luckygreen** [comment]
@noromamai the feature request hidden in the original question has been around for at least 6 years. I am pledging a reward of $100 or 100 Euro (implementer's choice) for someone to address this feature request once and for all.

Possible designs include:
* being able to put a PNG with a specific name into the same directory as the .qml and pulls in the PNG dynamically.
* a stand-alone script that is fed screensaver-main.qml and the .png, outputting a new screensaver-main.qml.
* something more elegant.

Design proposals have to be approved by me in advance of implementation to be eligible for the rewards. Proposals have to be received by me by 2025-02-09 (any time zone) to be eligible for the rewards.

Implementations need to be finished, checked in, and available to the end user by 2025-02-09 to receive the reward. If there is more than one implementation, I will choose the winner.

**Comment by aabytt** [comment]
@luckygreen do you mean the same bouncing logo screensaver but with an option to replace the image with your own?

**Comment by luckygreen** [comment]
> [@luckygreen](https://github.com/luckygreen) do you mean the same bouncing logo screensaver but with an option to replace the image with your own?

Correct. Many, if not most, users want to replace the bouncing DVD logo with a logo of their own. Without having to touch some base64-encoded string in some code on GitHub.

Here is one possible design that would work for an end user:
* put an image file of specs X onto a USB flash drive.
* Insert flash drive into LG TV.
* the code looks for an image file with a particular name in the root directory of any attached flash drives, sanity checks the file contents, and uses the file contents for the bouncing image.
* Implied in this design is a READ.ME that tells the user what the image file specs have to be. (In my case, I would want to be told the maximum pixel size and color depth of the image file. Others will need to know the minimum).
* if the code finds no such matching image file or there is no USB flash drive, display the default DVD logo.

There may be other designs that would work for an end user. I am not married to any particular design. But whatever the design, it can't require the end user to modify code on GitHub. That would not be end-user friendly.

**Comment by Anti-hero00** [comment]
I edited it a bit so you can just use pngs in a [fork](https://github.com/Anti-hero00/custom-screensaver). As I say in the readme I also got a different logo and cleaned it a bit as I didn't like the old one but you can use whatever the image just has to be like the original. you just need to download screensaver-main.qml and 300.png if you want to use that new logo from the assets folder and place it where I say in the readme. The size I would recommend is not too much bigger the 300X300 but its up to you really if you want something massive or small lol I didn't test it with massive pics. I assume where it is on my tv is the default location but probably won't be able to help if it isn't lol. I am hardly on here. I'm not really looking to fulfil luckygreen's request just thought it might help someone out in this issue and the other talking about the quality of the default image

**Comment by aarif2** [comment]
> Wehre can i add my custom images for this screen safer

You will have to edit the screensaver-main.qml file to anything you want. I have removed the moving DVD icon and set a background image with parallax effect. I also added a nice clock and date. There's still a lot of idea I want to try when I'll have free time again. Here's the preview of my current screensaver. 

![Image](https://github.com/user-attachments/assets/6b8cbcf2-4b35-4751-92e0-c4a9450d1974)

**Comment by exkc** [comment]
https://github.com/exkc/custom-screensaver/tree/main
i made one btw
it has a setting page :p
